### ISSUE TITLE
Questions about "Dataset size N and Fixed vertex size M"

### ISSUE THREAD

**Issue by K-Coconut** [davidireland3/LeNSE#4]
Hi, thank you for your impressive work.

I am confused about the hyperparameters $N$ and $M$, which is Dataset size and Fixed vertex size respectively in data generation phase.

I didn't find how you set these two parameters in the paper, and source code simply set $N$ as 5 (https://github.com/davidireland3/LeNSE/blob/31d117ced1cf7b4cf915c7cff29c343b37dedb2d/IM/fixed_size_dataset.py#L19)
and $M$ as 50 (https://github.com/davidireland3/LeNSE/blob/31d117ced1cf7b4cf915c7cff29c343b37dedb2d/IM/fixed_size_dataset.py#LL22C14-L22C14)

**Questions**
How many traning samples for each label do you use to train in your experiment?
And what is $M$ for different budget respectively, cause $M$ must be  greater than budget? $M=50$ in the source implementation should not work when budget is 100.

**Comment by davidireland-iso** [comment]
The information in the paper is found in the appendix, though with slightly different notation. In Table 3 I have provided the M, but you can define N = n * _num classes_. Most of the time we have _num classes_= 4, so N is usually 4 * n. But in a few cases we could not get a score low enough, so there were only 3 classes. Hopefully this clears things up. I think in the code you have sent as an example, n = 5 just to tune quickly to see if we can indeed find solutions of this size. In theory M could be equal to the budget, but this would require LeNSE to find exactly (or almost exactly) the correct solution. So yes, M should be larger than the budget but much smaller than the size of the original graph.

**Comment by K-Coconut** [comment]
Great, thank you for your replying.

**Comment by K-Coconut** [comment]
By the way, how to decide $M$, $N$ for each dataset and problem? If I have a new dataset on hand, how to set the parameters? Should I find the best value by grid search or following some rule?

**Comment by davidireland-iso** [comment]
To set $M$ it’s a bit tricky, because you want it to be large enough to solve easily but not so large that computational time is too much. So try and find a balance based on your compute budget. For $N$, I stuck with n * 4 but you can test using varying n with small M to see which best suits your score function and graph.
